First pass at DateInput (#1024)
hpe-design/design-system: aries-site/src/pages/components/dateinput.mdx

import { Example } from '../../layouts';
import {
  DateInputRangeExample,
  DateInputExample,
  DateInputSimpleExample,
  DateInputValidationExample,
} from '../../examples/components/dateinput';

<Example
  docs="https://v2.grommet.io/dateinput?theme=hpe"
  code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/dateinput/DateInputSimpleExample.js"
  figma="https://www.figma.com/file/Mf9if6zyGETTWW5b9PsYNQ/HPE-Date-Input-Component?node-id=1%3A567"
>
  <DateInputSimpleExample />
</Example>

## Guidance

- DateInput can be used to display past, present, or future dates and times.

> - DateInput is an intuitive way for users to input a date or time by providing them with a familiar calendar interface.

Reviewer: I think we can also say that DateInput provides a more guided experience for collecting dates from the user rather than a plain TextInput. The `format` provides MaskedInput constraints on what can be typed and auto-fills fixed characters. The Calendar Drop provides a familiar visual grid interface.

import { Example } from '../../layouts';
import {
  DateInputRangeExample,
  DateInputExample,
  DateInputSimpleExample,
  DateInputValidationExample,
} from '../../examples/components/dateinput';

<Example
  docs="https://v2.grommet.io/dateinput?theme=hpe"
  code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/dateinput/DateInputSimpleExample.js"

>   figma="https://www.figma.com/file/Mf9if6zyGETTWW5b9PsYNQ/HPE-Date-Input-Component?node-id=1%3A567"

Reviewer 2: ```suggestion
  figma="https://www.figma.com/file/Mf9if6zyGETTWW5b9PsYNQ/HPE-Date-Input-Component?node-id=1%3A2"
```

- The format provides MaskedInput constraints on what can be typed and auto-fills fixed characters. 
- The Calendar Drop provides a familiar visual grid interface.

### When using Date Input, you should:

- Use a label to instruct user what to do with the control.
- Labels should be clear and descriptive.
- Correctly format dates to reduce mistakes, such as having the caller specify
  the appropriate date format for the locale the user is in. 
- For example, in the U.S. dates are formatted as ‘mm/dd/yyyy’ whereas
 in other countries, such as Europe it might be ‘dd/mm/yyyy’.

## Variants
DateInput can be configured to support selection of a single date or a range of dates.

### Single Date

A single DateInput users can either manually input a date or select 
a date from the calendar

<Example
  docs="https://v2.grommet.io/dateinput?theme=hpe"
  code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/dateinput/DateInputExample.js"
  figma="https://www.figma.com/file/Mf9if6zyGETTWW5b9PsYNQ/HPE-Date-Input-Component?node-id=1%3A567"
>
  <DateInputExample />
</Example>

> ### Range Dates

Reviewer 2: ```suggestion
### Date Range
```

import { Example } from '../../layouts';
import {
  DateInputRangeExample,
  DateInputExample,
  DateInputSimpleExample,
  DateInputValidationExample,
} from '../../examples/components/dateinput';

<Example
  docs="https://v2.grommet.io/dateinput?theme=hpe"
  code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/dateinput/DateInputSimpleExample.js"
  figma="https://www.figma.com/file/Mf9if6zyGETTWW5b9PsYNQ/HPE-Date-Input-Component?node-id=1%3A567"
>
  <DateInputSimpleExample />
</Example>

## Guidance

> - DateInputs can be used to display past, present, or future dates or times.

Reviewer 3: ```suggestion
- DateInput can be used to display past, present, or future dates and times.
```

> - Use dateinput when you are asking the user for a time or date, or for scheduling tasks.

Reviewer 3: ```suggestion
- DateInput is an intuitive way for users to input a date or time by providing them with a familiar calendar interface.
```

> ## When using Date Input, you should:

Reviewer 3: ```suggestion
### When using Date Input, you should:
```

> * Use a label when appropriate

Reviewer 3: Accessibility question -- when/how would be the time to use DateInput without a label? Asking because I don't know the answer, not because I don't think it's a viable option.

Author: good question maybe we should change that sentence lol

> * Use a label when appropriate

Reviewer 3: ```suggestion
- Use a label when appropriate
```

> * Correctly format to reduce mistakes such as having the caller specify

Reviewer 3: ```suggestion
- Correctly format to reduce mistakes such as having the caller specify
```

  the appropriate date format for the locale the user is in. 

> * For example, in the U.S. dates are formatted as ‘mm/dd/yyyy’ whereas

Reviewer 3: ```suggestion
- For example, in the U.S. dates are formatted as ‘mm/dd/yyyy’ whereas
```

 in other countries, such as Europe it might be ‘dd/mm/yyyy’.

> ## Usage

Reviewer 3: ```suggestion
### Usage
```

> ### Usage

Reviewer 3: Feel like this "Usage" subheading isn't very necessary since this sentence was covered under "Guidance"

Reviewer 3: We could also remove the bullet below it

* Use the date input component when you want the user to assign past,
  present or future date(s).

## Variants

> - Date Input - This date input can used to pick one day or a range of dates,

Reviewer 3: maybe replace this paragraph with "DateInput can be configured to support selection of a single date or a range of dates."

depending on your use case. Concise labels should be included on this variant
to let the user know what to do.

### Single Date

A single DateInput users can either manually input a date or select 
a date from the calendar

<Example
  docs="https://v2.grommet.io/dateinput?theme=hpe"
  code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/dateinput/DateInputExample.js"
  figma="https://www.figma.com/file/Mf9if6zyGETTWW5b9PsYNQ/HPE-Date-Input-Component?node-id=1%3A567"
>
  <DateInputExample />
</Example>

### Range Dates

> Unlike a single date the RangeDate should be used for when a start and 

Reviewer 3: ```suggestion
DateInput can be configured to allow users to select a start and 
```

> end date is needed from the user. Users will need to use the calendar

Reviewer 3: ```suggestion
end date. Users will need to use the calendar
```

to select the range of dates.

<Example
  docs="https://v2.grommet.io/dateinput?theme=hpe"
  code="https://raw.githubusercontent.com/hpe-design/design-system/master/aries-site/src/examples/components/dateinput/DateInputRangeExample.js"
>
  <DateInputRangeExample />
</Example>

### Validation

> Invalid fields should be clear to the user what was missing

Reviewer 3: ```suggestion
Error messages should clearly identify when a DateInput was not correctly completed and what the user
```

> so the user can clearly understand which to address to fix the error.

Reviewer 3: ```suggestion
needs to do to fix the error.
```
